pkg/store/proxy_heap.go

```diff
@@ -111,24 +112,30 @@ func (d *dedupResponseHeap) Next() bool {
 func chainSeriesAndRemIdenticalChunks(series []*storepb.SeriesResponse) *storepb.SeriesResponse {
 	chunkDedupMap := map[uint64]*storepb.AggrChunk{}
 
+	buf := make([]byte, 8)
+	hasher := xxhash.New()
+
 	for _, s := range series {
 		for _, chk := range s.GetSeries().Chunks {
+			hasher.Reset()
 			for _, field := range []*storepb.Chunk{
 				chk.Raw, chk.Count, chk.Max, chk.Min, chk.Sum, chk.Counter,
 			} {
-				if field == nil {
-					continue
-				}
-				hash := field.Hash
-				if hash == 0 {
-					hash = xxhash.Sum64(field.Data)
-				}
-
-				if _, ok := chunkDedupMap[hash]; !ok {
-					chk := chk
-					chunkDedupMap[hash] = &chk
-					break
+				switch {
+				case field == nil:
+					binary.BigEndian.PutUint64(buf, 0)
+				case field.Hash != 0:
+					binary.BigEndian.PutUint64(buf, field.Hash)
+				default:
+					binary.BigEndian.PutUint64(buf, xxhash.Sum64(field.Data))
 				}
+				_, _ = hasher.Write(buf)
+			}
+			aggrChunkHash := hasher.Sum64()
```

Comment:
Do I get it right, that this actually calculates a hash of hashes of all the fields?
I wonder how it differs from calculating the hash of all the fields data in terms of possible hash collisions...

Reply:
Its the hash of field hashes ( mainly to reuse the field.Hash ).